src/python/zquantum/qcbm/target_thermal_states.py

```diff
@@ -0,0 +1,235 @@
+import bisect
+import typing
+import numpy as np
+from zquantum.core.bitstring_distribution import BitstringDistribution
+from zquantum.core.utils import dec2bin, bin2dec, convert_tuples_to_bitstrings
+
+
+def int2ising(number: int, length: int) -> typing.List[int]:
+    """Converts an integer into a +/-1s bitstring (also called Ising bitstring).
+    Args:
+        number: positive number to be converted into its corresponding Ising bitstring representation
+        length: length of the Ising bitstring (i.e. positive number of spins in the Ising system)
+    Returns:
+        Ising bitstring representation (1D array of +/-1)."""
+
+    binary_bitstring = dec2bin(number, length)
+    ising_bitstring = [bit * 2 - 1 for bit in binary_bitstring]
+    return ising_bitstring
+
+
+def ising2int(ising_bitstring: typing.List[int]) -> int:
+    """Converts a +/-1s bitstring (also called Ising bitstring) into an integer.
+    Args:
+        ising_bitstring: 1D array of +/-1.
+    Returns:
+        Integer number representation of the Ising bitstring."""
+
+    binary_bitstring = [int((bit + 1) / 2) for bit in ising_bitstring]
+    number = bin2dec(binary_bitstring)
+    return number
+
+
+def get_random_hamiltonian_parameters(n_spins: int) -> typing.Tuple:
+    """Generates random h, J, and where h and J are arrays of random coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd.
+    For reproducibilty, fix random generator seed in the higher level from which this function is called.
+    Args:
+        n_spins: positive number of spins in the Ising system.
+    Returns:
+       external_fields: n_spin coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd
+       two_body_couplings: n_spin x n_spin array of coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd
+    """
+    external_fields = np.zeros((n_spins))
+    two_body_couplings = np.zeros((n_spins, n_spins))
+    for i in range(n_spins):
+        external_fields[i] = np.random.normal(0, np.sqrt(n_spins))
+        for j in range(i, n_spins):
+            if i == j:
+                continue
+            two_body_couplings[i, j] = two_body_couplings[j, i] = np.random.normal(
+                0, np.sqrt(n_spins)
+            )
+    return external_fields, two_body_couplings
+
+
+def get_thermal_target_distribution_dict(
+    n_spins: int,
+    temperature: float,
+    hamiltonian_parameters: typing.List[np.array],  # TODO: add docstring
```

Comment:
Noting the TODO (I think it can be removed)

Reply:
Also, should `hamiltonian_parameters` be a tuple instead of a list since it has fixed size/types?

Reply:
I think the typing suggestion applies to other places in this PR too.

Reply:
Done!